Dpk updates

dpkingston opened the pull request:
There are a bunch of changes here, including some lint fixes (should have isolated those).
Added looping control, better and configurable handling of outbox, and added argurement parsing.
It would be nice to try and minimize the use of globals.  Perhaps putting that configuration information in a single, possibly global dict of parsed config.

dpkingston committed:
Add support for number of passes and fix a couple of errors.

dpkingston committed:
Added missing loop sleep.

bseidenberg commented on winlink_monitor.py:
```diff
@@ -76,29 +82,46 @@ def load_config(config_file):
+    global WINDOW_SIZE
     WINDOW_SIZE = int(config.get("health_window_size", 5))
 
+    # How many runs we look back at for determining health
```
Copy/paste needs updating

dpkingston replied:
Done.

bseidenberg commented on winlink_monitor.py:
```diff
@@ -76,29 +82,46 @@ def load_config(config_file):
     WINDOW_SIZE = int(config.get("health_window_size", 5))
 
+    # How many runs we look back at for determining health
+    global NEXT_PASS_DELAY
```
Please update config_syntax.md

bseidenberg commented on winlink_monitor.py:
```diff
@@ -76,29 +82,46 @@ def load_config(config_file):
+    #   from an aborted run, and remove it.
+    # False: if the mailbox being used is shared with a real user, then if we find
+    #   any mail there, be extra cautious and abort until the user clears the mail.
+    global DEDICATED_MAILBOX
```
Please update config_syntax.md

bseidenberg commented on winlink_monitor.py:
```diff
@@ -150,15 +172,21 @@ def load_config(config_file):
     for nodeobj in config.get("nodes", []):
         node = Node(nodeobj["name"], nodeobj["frequency"], nodeobj["peer"])
+        if args.nodes:
+            if node.name not in args.nodes and node.peer not in args.nodes:
+                continue
```
Should we throw an error for an invalid config? I generally prefer fail fast over silently drop.

dpkingston replied:
I am not checking the config.  I am seeing if the user supplied node identification matches something in our list.  The error would be if the user did not supply a node identification that was in our list.  They could easily typo the command line and this will take the safe route and do nothing.

bseidenberg commented on winlink_monitor.py:
```diff
@@ -269,7 +306,7 @@ def send_probe(node):
 def poll_for_probe(probe):
     sleep_int = FETCH_RETRY_INTERVAL
-    for i in range(FETCH_RETRIES_COUNT):
+    for _i in range(FETCH_RETRIES_COUNT):
```
Where's this from? Is this a python-ism I'm not aware of? 

Googling didn't turn anything up but suggests it may break ipython if the code is loaded there.

dpkingston replied:
I was actually in error here for a different reason.  And confused a bit.  _var is conventionally used for variables that should not be access from outside the context.  _ (alone) can be used for a variable that is intentionally going to be ignored (like a function return) and is intended to let lint and other users know that you are aware it returns a value and you are intentionally going to ignore the returned value.

dpkingston committed:
Update docs, fix error with earlier change (i/_i) which was erroneous anyways

dpkingston committed:
Fixed another bug.  Tested this time and its working as intended.

dpkingston committed:
Set powerlevel to medium (will make a config value later).  Lint fixes.

dpkingston committed:
Added -v/--verbose argument parsing to manage logging level.

dpkingston committed:
Convert comments to conventional docstrings.
https://peps.python.org/pep-0257/ for reference.

dpkingston committed:
Refactor of config information handling and some lint changes.
The big change here is to change from a bunch of globals to a single global
config dictionary.  The also lets us clean up the debugging print of config.
pylint recommended against using f-strings in logging so I fixed that.
Also changed how run() checking is handled and its now explicit on every invocation.

bseidenberg commented on tait.py:
```diff
@@ -218,11 +218,13 @@ def ccr_set_rx_ctcss(self, ctcss_tone_freq_hz):
-    
+
     At least one of the author's Tait radios has a nasty habit of not responding sometimes. So, we're going to do retries.
+    (We can't do exponential retries because modifying the read timeout requires closing/re-opening the serial device)
```
I removed this comment in my copy of the code, must have left it in the on-box copy. 

It's not true since there's a sleep after read (rather than during)

dpkingston replied:
Removed.

bseidenberg commented on tait.py:
```diff
@@ -241,26 +243,18 @@ def send_tait_cmd(self, cmd, arg_str):
             self.sp.write(bytes(msg, "utf-8"))
             # Return the response back to the caller
             ret = self.sp.read_until(b"\r")
-
```
I'd like to retain this (again, most have not made it to the on-box copy)

bseidenberg commented on winlink_monitor.py:
```diff
@@ -20,205 +20,217 @@
 #  * Logging needs to be cleaned up - and we need to figure out how to handle our output, pat's output, etc.
-#  * It's currently way too aggressive. It sends a message, sleeps 30 seconds, then downloads it from the internet, 
-#    then immediately moves to the next channel. The message send time is ~60s, so it's on-air ~66% of the time (and 
+#  * It's currently way too aggressive. It sends a message, sleeps 30 seconds, then downloads it from the internet,
```
I think you handled this.

dpkingston replied:
:-)

bseidenberg commented on winlink_monitor.py:
```diff
@@ -20,205 +20,217 @@
+        logging.basicConfig(level=logging.WARNING)
 
-def setup():
-    # FIXME
```
Thank you

dpkingston replied:
:-)

bseidenberg commented on winlink_monitor.py:
```diff
@@ -281,38 +301,43 @@ def poll_for_probe(probe):
-    # (OK... I just shelled out instead of writing it natively - exactly that)
-    output = run(f'grep -h Subject {MAILBOX_BASE}/in/* | cut -d : -f 2', shell=True, capture_output=True).stdout
+    '''Find all ids in inbox.
+       Basically grep Subject: $MAILBOX_DIR/* | cut -d : -f 2
+       (OK... I just shelled out instead of writing it natively - exactly that)
```
NP: These 2 lines are more of a comment than a docstring.

bseidenberg commented on winlink_monitor.py:
```diff
@@ -328,38 +353,49 @@ def history_string(history):
 def diff_and_report_health_state(old, new):
     for (node, health) in new.items():
-        if (old[node] != health):
-            # TODO - Real reporting
```
That TODO was my plan for tomorrow

(To be fair, I was just gonna dupe the logging line with a syslog() call.)

dpkingston replied:
I have been making the assumption that the calls to logging are being sent to syslog.  Lets just make sure that is true in setup (or earlier) with the potential to send to STDERR as well.

bseidenberg replied:
No, logging doesn't do syslog by default.

I was going to be more judicious with the syslog stuff than the rest of the logging.

bseidenberg reviewed (commented):
Few minor comments.

dpkingston commented on tait.py:
```diff
@@ -241,26 +243,18 @@ def send_tait_cmd(self, cmd, arg_str):
             # Return the response back to the caller
             ret = self.sp.read_until(b"\r")
-
-            # The most common failure mode we've seen is on the SACS UHF Tait:
-            #  A command will be sent and we will not get any reply back. 
-            #  If we retry the command, we get a checksum error, but it will work on the third try.
-            #  This probbaly indicates some sort of corruption on send (the newline doesn't make it through?)
-            #  or internal to the radio. 
```
Added back.

dpkingston reviewed (commented):
I think I have deal with your comments.  I'll research the logging configuration to make sure rsyslog is see the traffic (except debug)

dpkingston committed:
Addressing review comments.  Only impacts comments.

bseidenberg commented on winlink_monitor.py:
```diff
@@ -20,205 +20,214 @@
     try:
-        model_str = config['rig_model']
-        if (model_str == "TAIT"):
+        CONFIG['rig_port_speed'] = int(config_json['rig_port_speed'])
```
TODO: Document rig_port_speed in config_syntax.md

(Can be a follow-on PR)
